src/lib/currency/currency.ml

```diff
@@ -595,62 +596,146 @@ module Fee = struct
 end
 
 module Amount = struct
-  module T =
-    Make
-      (Unsigned_extended.UInt64)
-      (struct
-        let length = currency_length
-      end)
+  (* The basic type t of this module is defined in Mina_wire_types but is
```

Comment:
This comment will end up being repeated everywhere if we want to be thorough. Would it make more sense to move it into a top-level doc-comment in `Mina_wire_types`, and have a cross-reference to there each time?

Reply:
Yeah, I think I'll write a simple `mina_wire_types.mld` file with high-level documentation for the whole library.